### PR TITLE
Add GitHub Action for site and preview deployments

### PR DESCRIPTION
**Pull request by danielledeleo**
We now self-host the Unit docs site on Azure alongside other NGINX docs. This change adds a GitHub Action to build, preview, and deploy to the new environment. Some Netlify functionality, like previews has been replicated.

Related PR: https://github.com/nginxinc/kubernetes-ingress/pull/6027

Action Documentation: https://github.com/nginxinc/docs-actions/tree/main?tab=readme-ov-file#docs-actions

### CODE REVIEW

**Comment by danielledeleo** [comment]
I won't be able to test this action until it is merged (🐓🥚). Merging it will not affect the live site, as deployments to production must be done by running the workflow manually.

File: .github/workflows/docs-build-push.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        options:
+        - preview
+        - dev
+        - staging
+        - prod
+  pull_request:
+    branches:
+      - "*"
+    paths:
+      - "docs/**"
```
Comment (javorszky) [inline .github/workflows/docs-build-push.yml:19]: given that the unit-docs repo is not like the rest of the documentation that live together with the code, and in this repository we don't have a `docs` folder, is this still okay to leave here, or should this change?

Or will this be relevant once we actually migrate from rst -> hugo?

Comment (danielledeleo) [reply .github/workflows/docs-build-push.yml:19]: Good catch, thanks.

**Comment by callahad** [comment]
Concerned about dropping the redirects / url proxying from netlify.toml without the commit message saying anything about what's happened to them.

...also a bit nervous about the record of those disappearing from this repo; knowing to redirect `/download` and transparently proxy `/download/*` seems pretty crucial to making the site actually work.

**Comment by nginx-jack** [comment]
> Concerned about dropping the redirects / url proxying from netlify.toml without the commit message saying anything about what's happened to them.
> 
> ...also a bit nervous about the record of those disappearing from this repo; knowing to redirect `/download` and transparently proxy `/download/*` seems pretty crucial to making the site actually work.

We can add a commit message explaining that those redirects now live in a private repository as an NGINX configuration, and add something about it to a readme as well, so if/when unit changes hosting again, we make sure we don't lose those redirects.

I'd like to get to a point, like netlify, where each documentation resource can provide it's own NGINX snippet containing it's required proxying and redirects, but we're not there yet unfortunately.

**Review by avahahn — APPROVED** [review]

**Pull request merged**